addons/product/models/product_pricelist.py

```diff
@@ -333,6 +333,38 @@ def _get_partner_pricelist(self, partner_id, company_id=None):
 
         return pl
 
+    def _get_partner_pricelist_multi(self, partner_ids, company_id=None):
+        """ Retrieve the applicable pricelist for given partners in a given company.
+
+            :param company_id: if passed, used for looking up properties,
+                instead of current user's company
+            :return: a dict {partner_id: pricelist}
+        """
+        Partner = self.env['res.partner']
+        Property = self.env['ir.property'].with_context(force_company=company_id or self.env.user.company_id.id)
+
+        # retrieve values of property
+        result = Property.get_multi('property_product_pricelist', Partner._name, partner_ids)
+
+        # group partners without value, and find a pricelist for each group
+        remaining_partner_ids = [pid for pid, val in result.items() if not val]
+        domain = [('id', 'in', remaining_partner_ids)]
+        groups = Partner.read_group(domain, ['country_id'], ['country_id'])
+        for group in groups:
+            country_id = group['country_id'] and group['country_id'][0]
+            pl = self.env['product.pricelist'].search([('country_group_ids.country_ids', '=', country_id)], limit=1)
+
+            if not pl:
+                pl = Property.get('property_product_pricelist', 'res.partner')
+
+            if not pl:
+                pl = self.env['product.pricelist'].search([], limit=1)
```

Comment:
Since we love letting users do whatever they want, it is possible for this to return `None`, what happens in this case? Should we log it? Raise an error?

Reply:
Indeed, if there is no pricelist, then the value of the property is falsy...